src/SentryLink.ts

```diff
@@ -1,225 +1,113 @@
-import { Scope, Severity } from '@sentry/types';
-import deepMerge from 'deepmerge';
+import {
+  ApolloLink,
+  FetchResult,
+  NextLink,
+  Operation,
+} from '@apollo/client/core';
+import { Severity } from '@sentry/types';
 import Observable from 'zen-observable';
 
-import { FetchResult } from '@apollo/client/link/core/types';
-
+import { GraphQLBreadcrumb, makeBreadcrumb } from './breadcrumb';
+import { ApolloLinkSentryOptions, Options, withDefaults } from './options';
 import {
-  ApolloLink, NextLink, Operation as ApolloOperation,
-} from '@apollo/client/link/core';
-
-import { addBreadcrumb, configureScope } from '@sentry/minimal';
-import { OperationBreadcrumb } from './OperationBreadcrumb';
-import { Operation } from './Operation';
-
-export interface Options {
-  setTransaction?: boolean;
-  setFingerprint?: boolean;
-
-  breadcrumb?: {
-    enable?: boolean;
-    includeQuery?: boolean;
-    includeCache?: boolean;
-    includeVariables?: boolean;
-    includeResponse?: boolean;
-    includeError?: boolean;
-    includeContextKeys?: string[];
-  }
-
-  filter?: (operation: Operation) => boolean;
-  beforeBreadcrumb?: (breadcrumb: OperationBreadcrumb) => OperationBreadcrumb;
-}
-
-const defaultOptions: Options = {
-  setTransaction: true,
-  setFingerprint: true,
-
-  breadcrumb: {
-    enable: true,
-    includeQuery: false,
-    includeCache: false,
-    includeVariables: false,
-    includeResponse: false,
-    includeError: false,
-    includeContextKeys: [],
-  },
-};
+  attachBreadcrumbToSentry,
+  setFingerprint,
+  setTransaction,
+} from './sentry';
 
 export class SentryLink extends ApolloLink {
-  private readonly options: Options;
+  private readonly options: ApolloLinkSentryOptions;
 
-  /**
-   * Create a new ApolloLinkSentry
-   * @param {Options} options
-   */
   constructor(options: Options = {}) {
     super();
-    this.options = deepMerge(defaultOptions, options);
+    this.options = withDefaults(options);
   }
 
-  /**
-   * This is where the GraphQL operation is received
-   * A breadcrumb will be created for the operation, and error/response data will be handled
-   * @param {ApolloOperation} op
-   * @param {NextLink} forward
-   * @returns {Observable<FetchResult> | null}
-   */
-  request = (op: ApolloOperation, forward: NextLink): Observable<FetchResult> | null => {
-    // Obtain necessary data from the operation
-    const operation = new Operation(op);
-
-    // Create a new breadcrumb for this specific operation
-    const breadcrumb = new OperationBreadcrumb();
-    this.fillBreadcrumb(breadcrumb, operation);
-
-    // Start observing the operation for results
-    return new Observable<FetchResult>((observer) => {
```

Comment:
I thought this wrapping was totally unnecessary and ripped it out. Turns out it is quite important to guarantee the callbacks run before the error actually hits Sentry. I now commented that section heavily 😄